### PR TITLE
Add lock barriers

### PR DESCRIPTION
**Pull request by aaannz**
[merge only after #838 and subsequent rebase]

- POST /api/v1/barrier
  - creates new barrier, takes mandatory form parameters
    - name - name of the barrier
    - tasks - number of tasks to wait for
- POST /api/v1/barrier/:barrier_name
  - register itself as waiting for the barrier to be released
- DELETE /api/v1/barrier/:barrier_name
  - destroy existing barrier

### CODE REVIEW

**Comment by coolo** [comment]
this should make the HA tests easier to parse (once we have mmapi support that is :)

**Comment by okurz** [comment]
+1

File: lib/OpenQA/Schema/Result/Comments.pm [hunk 1]
```diff
@@ -42,8 +42,9 @@ __PACKAGE__->add_columns(
         data_type      => 'integer',
         is_foreign_key => 1,
     },
-    hidden => {
-        data_type     => 'boolean',
+    flags => {
```
Comment (coolo) [inline lib/OpenQA/Schema/Result/Comments.pm]: how is this related? and shouldn't be there a migration of the data?

Comment (aaannz) [inline lib/OpenQA/Schema/Result/Comments.pm]: Not related at all, I wanted to do one db schema change instead of multiple. I didn't find any place where we use this "hidden" column, that's why I didn't bother with migration. Did I miss someplace?

Comment (coolo) [inline lib/OpenQA/Schema/Result/Comments.pm]: But I can't find any reasoning either

Comment (aaannz) [inline lib/OpenQA/Schema/Result/Comments.pm]: I don't get what do you mean.

Comment (coolo) [inline lib/OpenQA/Schema/Result/Comments.pm]: I don't see any 'why' in your commits

Comment (aaannz) [inline lib/OpenQA/Schema/Result/Comments.pm]: It's what we discussed in #837

Thing to consider though is if it is better to just add pinned column and have separate column for each feature or do it old school with multiple flags in one column.

Comment (coolo) [inline lib/OpenQA/Schema/Result/Comments.pm]: Might be - but I don't think we will be able to find the discussion in #837 when looking for a why

Comment (coolo) [inline lib/OpenQA/Schema/Result/Comments.pm]: (just as I wasn't able to find it, as I didn't follow #837)

Comment (aaannz) [inline lib/OpenQA/Schema/Result/Comments.pm]: Indeed, I'll update commit message as soon as I'll get to some computer with actual keyboard.

Comment (aaannz) [inline lib/OpenQA/Schema/Result/Comments.pm]: Commit message updated

**Comment by okurz** [comment]
@aaannz After talking to @Martchus today I am not sure if flags for comments is the right approach or if we rather need description fields for each testsuite, job group, etc. IMHO for this database migrate consider only what's actually used now. Anyway, @Martchus can now rely on the commits you proposed and experiment what works best.

**Comment by aaannz** [comment]
I wanted to go with flags in original #837, so if it were by me, flags are actually used now. The deal was I'll add the flags so when we'll enhance comments next time, we don't need to use hacks.

Speaking of hacks though ( :) ) I think this persistent metadata (desc, maintainer, ...) can be put into TestSuite settings. The similar for workers, products. We can then use common helper which will look for special variables, e.g. __desc, __main, etc. Filter them out for workers, display in special way in webui.
Still need to add something like JobGroupSettings for JobGroups though.

**Comment by okurz** [comment]
yes, this is what I had in mind, too. Thinking about where descriptions could make sense we came to the idea that each test_suite knows a description and then this could be displayed on each job using that test_suite. This can't be easily done over comments which belong to jobs.
